blocks/library/embed/index.js

```diff
@@ -114,7 +117,12 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 							return;
 						}
 						response.json().then( ( obj ) => {
-							const { html, type } = obj;
+							const { html } = obj;
+							let { type } = obj;
+
+							if ( includes( html, 'class="wp-embedded-content" data-secret' ) ) {
```

Comment:
If the external WordPress site returns a response not containing the 'data-secret' attribute (eg: previous version of WordPress once the code we temporarily added in lib/compat.php will be included in WordPress core), then the response is treated like any other embed content.